rs/registry/canister/src/mutations/node_management/common.rs

```diff
@@ -226,6 +226,13 @@ fn get_key_family<T: prost::Message + Default>(
     registry: &Registry,
     prefix: &str,
 ) -> Vec<(String, T)> {
+    get_key_family_iter(registry, prefix).collect()
+}
+
+pub(crate) fn get_key_family_iter<T: prost::Message + Default>(
```

Comment:
new method to more efficiently iterate over all entries of a particular type than what we were doing, but without the memory overhead of collecting them into a vector all at once.